File: python/src/commands/CommandRegistry.py
from typing import List
from src.commands.ICommand import ICommand


class CommandRegistry:
    _commands = {}

    def registerCommand(self, commandKeyword: str, command: ICommand) -> None:
        self._commands[commandKeyword] = command

    def unRegisterCommand(self, commandKeyword: str) -> None:
        self._commands.pop(commandKeyword)

    def _get(self, commandName: str) -> ICommand:
        return self._commands.get(commandName)

    def _parse(self, input: str) -> List[str]:
        return input.strip().split(" ")

    def invokeCommand(self, input: str) -> None:
        tokens = self._parse(input)
        command = self._get(tokens[0])
        if command is None:
            raise Exception("INVALID COMMAND 🛑")
        command.invoke(tokens)

File: java/src/main/java/com/example/demo/commands/CommandRegistry.java
package com.example.demo.commands;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public class CommandRegistry {
    private static final Map<String, ICommand> commands = new HashMap<>();

    public void registerCommand(String commandKeyword, ICommand command) {
        commands.putIfAbsent(commandKeyword,command);
    }

    public void unRegisterCommand(String commandKeyword) {
        commands.remove(commandKeyword);
    }

    private ICommand get(String commandName){
        return commands.get(commandName);
    }

    private List<String> parse(String input){
        return Arrays.stream(input.split(" ")).collect(Collectors.toList());
    }


    public void invokeCommand(String input) {
        List<String> tokens = parse(input);
        ICommand command = get(tokens.get(0));
        if(command == null){
            // Handle Exception
            throw new RuntimeException("INVALID COMMAND 🛑");
        } 
        command.invoke(tokens);
        return;
    }
}
